packages/apps/composer-app/src/plugins/welcome/credentials/util.ts

```diff
@@ -54,3 +54,56 @@ export const activateAccount = async ({
   const { credential } = await response.json();
   return credential && codec.decode<Credential>(credential);
 };
+
+type ProfileResponse = {
+  identityDid: string;
+  email?: string;
+  capabilities: string[];
+};
+
+/**
+ * Get profile.
+ *
+ * @param params.hubUrl
+ * @param params.credential
+ */
+export const getProfile = async ({
+  hubUrl,
+  credential,
+}: {
+  hubUrl: string;
+  credential: Credential;
+}): Promise<ProfileResponse> => {
+  const response = await fetch(new URL('/account/profile', hubUrl), {
```

Comment:
(generically), if we're abstracting the request/response, then return the profile, not the ProfileResponse?